zkvm/docs/zkvm-spec.md

```diff
@@ -828,6 +787,54 @@ V == v·B + 0·B2
+
+
+### Call proof
+
+This struct is an argument to the [`call`](#call) instruction. It contains the information to verify that a program is represented in the [Taproot](#taproot) Merkle tree, in the following fields:
+ - `X`, the signing key
+ - `neighbors`, an array of left-right neighbors in the Merkle tree. Each is a LE32 item representing the hash of its children. It is the neighbor of an item on the Merkle path to the leaf hash of the program string. The leaf hash and root hash are not included. Neighbor hashes are ordered from bottom-most to top-most. As an example, consider a Merkle tree with more than three levels. The first element is the neighbor of the leaf hash; the second element is the neighbor of the leaf hash's parent; and the last is a child of the Merkle root.
+ - `positions`, a pattern indicating each neighbor's position. 0 represents a left neighbor, and 1 represents a right neighbor. This is a 32-bit little-endian integer, where the lower bits indicate the position of the lower-level neighbors. These bits are ordered identically to `neighbors`, because there is a one-to-one correspondence between their elements. Consider the example in the previous paragraph again. The position of the leaf hash's neighbor is represented by the first bit; the position of the neighbor of the leaf hash's parent is represented by the second bit; and the position of a child of the Merkle root is the last bit.
+
+![An example Merkle tree to demonstrate the call proof.](taproot_example.svg)
+
+Here is a visual example using the tree above. Suppose we are verifying the program with the leaf hash `C`, with a right neighbor `D`. The parent of `C` is `J = h(C || D)`, with a left neighbor `I`. The parent of `J` is `M = h(I || J)`, with a right neighbor `N`. `M`'s parent is the Merkle root `R`.
+This gives the neighbor list [`D`, `I`, `N`] and bit pattern `101`. The Merkle path is red, while the neighbors are blue.
+
+
+
+### Blinded program
+
+The blinded program is used to construct the [Taproot tree](#taproot). It is an enum that can either be a [program](#program) or its accompanying blinding factor. A leaf element in the Taproot tree is a hash of a blinded program. This lets us store the hash of the blinding factor in the tree: each odd leaf will be the hash of a program, and each even leaf a hash of the program's blinding factor.
```

Comment:
actually randomizing the relative positions of the blinding factor and the program is better for privacy, but we can tweak this as a follow-up